.gitattributes

```diff
@@ -1,3 +1,40 @@
-* text=auto eol=lf
+# Set default behavior to automatically normalize line endings.
+* whitespace=!indent,trail,space text=auto
+
+# Force Windows batch scripts to always use CRLF line endings so that if a repo
+# is accessed in Windows via a file share from *nix, the scripts still work.
 *.{cmd,[cC][mM][dD]} text eol=crlf
 *.{bat,[bB][aA][tT]} text eol=crlf
+
+# Force amenable shell scripts always to use LF line endings so that if a repo
+# is accessed in *nix via a file share from Windows, the scripts still work.
```

Comment:
```suggestion
# Force amenable shell scripts always to use LF line endings so that the scripts
# still work if a repo is accessed under *nix via a file share from Windows.
```